server/game/baseability.js

```diff
@@ -60,7 +60,20 @@ class BaseAbility {
      * @returns {Boolean}
      */
     canPayCosts(context) {
-        return _.all(this.cost, cost => cost.canPay(context));
+        return this.withContext(context, 'cost', () => _.all(this.cost, cost => cost.canPay(context)));
+    }
+
+    withContext(context, stage, callback) {
```

Comment:
What 'withContext'?  (Naming Is Hard, I know).  This method probably needs to state what it's doing with the context (executeWithContext, resolveWithContext etc)

Reply:
Renamed.

Reply:
Thanks, much clearer!